Lib/asyncio/queues.py

```diff
@@ -214,6 +254,33 @@ async def join(self):
         if self._unfinished_tasks > 0:
             await self._finished.wait()
 
+    def shutdown(self, immediate=False):
+        """Shut-down the queue, making queue gets and puts raise
+        QueueShutDown.
+
+        By default, gets will only raise once the queue is empty. Set
+        'immediate' to True to make gets raise immediately instead.
+
+        All blocked callers of put() will be unblocked, and also get()
+        and join() if 'immediate'.
+        """
+        self._is_shutdown = True
+        if immediate:
+            while not self.empty():
+                self._get()
+                if self._unfinished_tasks > 0:
+                    self._unfinished_tasks -= 1
+            if self._unfinished_tasks <= 0:
```

Comment:
Not that it makes one iota of difference, but comparing this to the code in `task_done()` this could be `==`?
```suggestion
            if self._unfinished_tasks == 0:
```

Reply:
Yeah, I think `==` makes more sense since a negative value for `_unfinished_tasks` would signal something went wrong.

Reply:
This is a hold-over from the [multi-threaded implementation](https://github.com/python/cpython/pull/104750#discussion_r1483950194), and is not needed anymore